The extension pointing at wrong vscode config folder in open source build of VSCode (Arch/Manjaro)

- [The Open Source build of Visual Studio Code (AUR)](https://aur.archlinux.org/packages/code/)

- VSCode version:
```
Code - OSS
Version 1.24.1
Commit 24f62626b222e9a8313213fb64b10d741a326288
Date 2018-06-25T07:36:31.114Z
Shell 1.7.12
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64
```

- Git Project Manager version: `1.6.1`

- Operating System:
```
Manjaro
$ uname -a
Linux ngekngok 4.14.48-2-MANJARO #1 SMP PREEMPT Fri Jun 8 20:41:40 UTC 2018 x86_64 GNU/Linux
```

- Developer Tools > Console Log:
```
console.ts:136 [Extension Host] WARNING: Promise with no error callback:7
console.ts:136 [Extension Host]% Object {exception: Object, error: null, promise: Object, id: 7}error: nullexception: Objectid: 7promise: Object__proto__: Object
console.ts:136 [Extension Host] Error: ENOENT: no such file or directory, open '/home/schadenfreude/.config/Code/User/gpm-recentItems.json'
	at Object.fs.openSync (fs.js:584:18)
	at Object.module.(anonymous function) [as openSync] (ELECTRON_ASAR.js:173:20)
	at Object.fs.writeFileSync (fs.js:1326:33)
	at RecentItems.saveToFile (/home/schadenfreude/.vscode-oss/extensions/felipecaputo.git-project-manager-1.6.1/src/recentItems.js:34:12)
	at RecentItems.addProject (/home/schadenfreude/.vscode-oss/extensions/felipecaputo.git-project-manager-1.6.1/src/recentItems.js:51:14)
	at GitProjectManager.openProject (/home/schadenfreude/.vscode-oss/extensions/felipecaputo.git-project-manager-1.6.1/src/gitProjectManager.js:250:25)
	at onResolve (/home/schadenfreude/.vscode-oss/extensions/felipecaputo.git-project-manager-1.6.1/src/gitProjectManager.js:161:22)
	at Object.g [as _notify] (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:76:971)
	at Object.enter (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:80:277)
	at n.Class.derive._oncancel._run (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:82:84)
	at n.Class.derive._oncancel._completed (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:81:523)
	at /usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:103:304
	at n.Class.derive._creator.done (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:83:385)
	at Object.g [as _notify] (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:77:94)
	at Object.enter (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:80:277)
	at n.Class.derive._oncancel._run (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:82:84)
	at n.Class.derive._oncancel._completed (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:81:523)
	at e.resolveOk (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:545:791)
	at e._receiveReply (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:636:165)
	at e._receiveOneMessage (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:635:473)
	at /usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:634:315
	at /usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:637:395
	at /usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:95:432
	at e.fire (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:96:764)
	at Socket.<anonymous> (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:154:338)
	at emitOne (events.js:96:13)
	at Socket.emit (events.js:191:7)
	at readableAddChunk (_stream_readable.js:178:18)
	at Socket.Readable.push (_stream_readable.js:136:10)
	at Pipe.onread (net.js:560:20)
mainThreadExtensionService.ts:43 [felipecaputo.git-project-manager]ENOENT: no such file or directory, open '/home/schadenfreude/.config/Code/User/gpm-recentItems.json'
e.$onExtensionRuntimeError @ mainThreadExtensionService.ts:43
e._doInvokeHandler @ rpcProtocol.ts:265
e._invokeHandler @ rpcProtocol.ts:250
e._receiveRequest @ rpcProtocol.ts:196
e._receiveOneMessage @ rpcProtocol.ts:178
(anonymous) @ rpcProtocol.ts:111
(anonymous) @ rpcProtocol.ts:307
e.fire @ event.ts:140
(anonymous) @ ipc.net.ts:81
emitOne @ events.js:96
emit @ events.js:191
readableAddChunk @ _stream_readable.js:178
Readable.push @ _stream_readable.js:136
onread @ net.js:560
mainThreadExtensionService.ts:44 Error: ENOENT: no such file or directory, open '/home/schadenfreude/.config/Code/User/gpm-recentItems.json'
	at Object.fs.openSync (fs.js:584:18)
	at Object.module.(anonymous function) [as openSync] (ELECTRON_ASAR.js:173:20)
	at Object.fs.writeFileSync (fs.js:1326:33)
	at RecentItems.saveToFile (/home/schadenfreude/.vscode-oss/extensions/felipecaputo.git-project-manager-1.6.1/src/recentItems.js:34:12)
	at RecentItems.addProject (/home/schadenfreude/.vscode-oss/extensions/felipecaputo.git-project-manager-1.6.1/src/recentItems.js:51:14)
	at GitProjectManager.openProject (/home/schadenfreude/.vscode-oss/extensions/felipecaputo.git-project-manager-1.6.1/src/gitProjectManager.js:250:25)
	at onResolve (/home/schadenfreude/.vscode-oss/extensions/felipecaputo.git-project-manager-1.6.1/src/gitProjectManager.js:161:22)
	at Object.g [as _notify] (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:76:971)
	at Object.enter (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:80:277)
	at n.Class.derive._oncancel._run (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:82:84)
	at n.Class.derive._oncancel._completed (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:81:523)
	at /usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:103:304
	at n.Class.derive._creator.done (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:83:385)
	at Object.g [as _notify] (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:77:94)
	at Object.enter (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:80:277)
	at n.Class.derive._oncancel._run (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:82:84)
	at n.Class.derive._oncancel._completed (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:81:523)
	at e.resolveOk (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:545:791)
	at e._receiveReply (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:636:165)
	at e._receiveOneMessage (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:635:473)
	at /usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:634:315
	at /usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:637:395
	at /usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:95:432
	at e.fire (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:96:764)
	at Socket.<anonymous> (/usr/share/code-oss/resources/app/out/vs/workbench/node/extensionHostProcess.js:154:338)
	at emitOne (events.js:96:13)
	at Socket.emit (events.js:191:7)
	at readableAddChunk (_stream_readable.js:178:18)
	at Socket.Readable.push (_stream_readable.js:136:10)
	at Pipe.onread (net.js:560:20)
e.$onExtensionRuntimeError @ mainThreadExtensionService.ts:44
e._doInvokeHandler @ rpcProtocol.ts:265
e._invokeHandler @ rpcProtocol.ts:250
e._receiveRequest @ rpcProtocol.ts:196
e._receiveOneMessage @ rpcProtocol.ts:178
(anonymous) @ rpcProtocol.ts:111
(anonymous) @ rpcProtocol.ts:307
e.fire @ event.ts:140
(anonymous) @ ipc.net.ts:81
emitOne @ events.js:96
emit @ events.js:191
readableAddChunk @ _stream_readable.js:178
Readable.push @ _stream_readable.js:136
onread @ net.js:560
vscode.env.appName
VM286:1 Uncaught ReferenceError: vscode is not defined
    at <anonymous>:1:1
(anonymous) @ VM286:1
```

- `~/.config/` directory:
```
$ ls ~/.config | grep Code
Code - OSS
```

- `~/.config/Code - OSS/User/` directory:
```
$ ls ~/.config/Code\ -\ OSS/User
settings.json  snippets  vsicons.settings.json  workspaceStorage
```

- Similar issue in other extension:
Project Manager: issue [#178](https://github.com/alefragnani/vscode-project-manager/issues/178) (fixed in [#179](https://github.com/alefragnani/vscode-project-manager/pull/179))

- Another console log:
```
/usr/share/code-oss/resources/app/out/vs/workbench/workbench.main.js:1311  INFO no standard startup: not just one text editor
/usr/share/code-oss/resources/app/out/vs/workbench/workbench.main.js:268 [Extension Host] vscode-icons is active!
/usr/share/code-oss/resources/app/out/vs/workbench/workbench.main.js:268 [Extension Host] rejected promise not handled within 1 second
t.log @ /usr/share/code-oss/resources/app/out/vs/workbench/workbench.main.js:268
/usr/share/code-oss/resources/app/out/vs/workbench/workbench.main.js:2960 [felipecaputo.git-project-manager]ENOENT: no such file or directory, open '/home/schadenfreude/.config/Code/User/gpm_projects.json'
e.$onExtensionRuntimeError @ /usr/share/code-oss/resources/app/out/vs/workbench/workbench.main.js:2960
/usr/share/code-oss/resources/app/out/vs/workbench/workbench.main.js:2960 Error: ENOENT: no such file or directory, open '/home/schadenfreude/.config/Code/User/gpm_projects.json'
	at Object.fs.openSync (fs.js:584:18)
	at Object.module.(anonymous function) [as openSync] (ELECTRON_ASAR.js:173:20)
	at Object.fs.writeFileSync (fs.js:1326:33)
	at GitProjectManager.saveListToDisc (/home/schadenfreude/.vscode-oss/extensions/felipecaputo.git-project-manager-1.6.1/src/gitProjectManager.js:69:12)
	at GitProjectManager.saveRepositoryInfo (/home/schadenfreude/.vscode-oss/extensions/felipecaputo.git-project-manager-1.6.1/src/gitProjectManager.js:95:14)
	at GitProjectManager.updateRepoList (/home/schadenfreude/.vscode-oss/extensions/felipecaputo.git-project-manager-1.6.1/src/gitProjectManager.js:193:14)
	at projectLocator.locateGitProjects.then.then.dirList (/home/schadenfreude/.vscode-oss/extensions/felipecaputo.git-project-manager-1.6.1/src/gitProjectManager.js:218:43)
	at <anonymous>
e.$onExtensionRuntimeError @ /usr/share/code-oss/resources/app/out/vs/workbench/workbench.main.js:2960
```

--- a/src/gitProjectManager.js
+++ b/src/gitProjectManager.js
@@ -383,11 +383,7 @@ class GitProjectManager {
         });
     }
     getChannelPath() {
-        if (vscode.env.appName.indexOf("Insiders") > 0) {
-            return "Code - Insiders";
-        } else {
-            return "Code";
-        }
+        return vscode.env.appName.replace("Visual Studio ", "");
     }
 
 }

